modules/saml/lib/IdP/SAML2.php

```diff
@@ -251,6 +253,26 @@ private static function getAssertionConsumerService(
             return $firstFalse;
         }
 
+        $skipEndpointValidation = false;
+        if ($authnRequestSigned === true) {         
+            $skipEndpointValidationWhenSigned = $spMetadata->getValue('skipEndpointValidationWhenSigned', false);
+            if (is_bool($skipEndpointValidationWhenSigned) === true) {
+                $skipEndpointValidation = $skipEndpointValidationWhenSigned;
+            } elseif (is_callable($skipEndpointValidationWhenSigned) === true) {
```

Comment:
I purposely made this a callable because I wanted the ability to make the option dependent on other things in the metadata (like the existence of entity attributes). 

I maintain an internal federation of multiple IdPs and SPs with disparate implementations of SAML.  Some of these require the IdP to respond to the ACS url they send in the AuthnRequest (which is why I felt the need to add this feature).  I aggregate and build and a metadata file for this mini-federation and use the metarefresh module to pull it into SSP.  However, the metarefresh module doesn't provide the ability to customize/change the generated metadata after it is "translated" from XML.  You can inject "template" attributes, but again - that cannot be dependent on other things about the metadata, which is what I need.  My SPs that require this feature announce it via an entity attribute and I wanted to check for that and make this option dependent on its presence.